### PR TITLE
android:textFontWeight not taking effect on AppCompatTextView

### PR DESCRIPTION
**Pull request by simplepeng**
## Proposed Changes

Resolving the issue of android:textFontWeight not taking effect on AppCompatTextView when typeface or fontFamily is not set.

## Testing

AppCompatTextViewTest Test: testTextFontWeight

### CODE REVIEW

**Comment by simplepeng** [comment]
Could someone help with reviewing this?
@yigit @liutikas @dlam @ianhanniballake

**Comment by simplepeng** [comment]
https://issuetracker.google.com/u/1/issues/368087671

**Review by nona-google — APPROVED** [review]
lgtm

**Comment by liutikas** [comment]
we get the following crash when running the test on API 21 and 26
```
java.lang.NoSuchMethodError: No virtual method getWeight()I in class Landroid/graphics/Typeface; or its super classes (declaration of 'android.graphics.Typeface' appears in /system/framework/framework.jar)
	at androidx.appcompat.widget.AppCompatTextViewTest.testTextFontWeight(AppCompatTextViewTest.java:1307)
	```

File: appcompat/appcompat/src/androidTest/java/androidx/appcompat/widget/AppCompatTextViewTest.java [hunk 1]
```diff
@@ -1299,6 +1299,10 @@ public void onDestroyActionMode(ActionMode mode) {
     @RequiresApi(api = Build.VERSION_CODES.P)
```
Comment (liutikas) [inline appcompat/appcompat/src/androidTest/java/androidx/appcompat/widget/AppCompatTextViewTest.java]: You should use `@SdkSupress(minSdkVersion=` here

Comment (simplepeng) [reply appcompat/appcompat/src/androidTest/java/androidx/appcompat/widget/AppCompatTextViewTest.java]: thanks , i got it.

**Pull request closed without merging**